components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/listener/ClaimValueEncryptionListener.java

```diff
@@ -0,0 +1,239 @@
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.common.AbstractUserStoreManager;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ClaimValueEncryptionListener extends AbstractIdentityUserOperationEventListener {
+
+    private static final Log LOG = LogFactory.getLog(ClaimValueEncryptionListener.class);
+
+    private static final String CLAIM_VALUE = "ClaimValue";
+    private static final String CLAIM_URI = "claimURI";
+
+    @Override
+    public int getExecutionOrderId() {
+
+        int orderId = getOrderId();
+        if (orderId != IdentityCoreConstants.EVENT_LISTENER_ORDER_ID) {
+            return orderId;
+        }
+        return 94;
+    }
+
+    @Override
+    public boolean doPreAddUserWithID(String userID, Object credential, String[] roleList, Map<String, String> claims,
+                                      String profile, UserStoreManager userStoreManager) throws UserStoreException {
+
+        try {
+            if (!isEnable() || userStoreManager == null) {
+                return true;
+            }
+
+            for (Map.Entry<String, String> entry : claims.entrySet()) {
+                String claimURI = entry.getKey();
+                if (checkEnableEncryption(claimURI, userStoreManager)) {
+                    String claimValue = entry.getValue();
+                    try {
+                        claimValue = encryptClaimValue(claimValue);
+                    } catch (CryptoException e) {
+                        LOG.error("Error occurred while encrypting claim value", e);
+                    }
+                    claims.put(claimURI, claimValue);
+                }
+            }
+            return true;
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            throw new UserStoreException("Error while retrieving tenant ID", e);
+        }
+    }
+
+    @Override
+    public boolean doPreSetUserClaimValues(String userName, Map<String, String> claims, String
+            profileName, UserStoreManager userStoreManager) throws UserStoreException {
+
+        if (!isEnable() || userStoreManager == null) {
+            return true;
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Pre set claims is called in ClaimValueEncryptionListener");
+        }
+
+        for (Map.Entry<String, String> entry : claims.entrySet()) {
+            String claimURI = entry.getKey();
+            if (checkEnableEncryption(claimURI, userStoreManager)) {
+                String claimValue = entry.getValue();
+                try {
+                    claimValue = encryptClaimValue(claimValue);
+                } catch (CryptoException e) {
+                    LOG.error("Error occurred while encrypting claim value", e);
+                }
+                claims.put(claimURI, claimValue);
+            }
+        }
```

Comment:
seems the code is duplicated in above method, can you extract this to a common method and eliminate the duplication?